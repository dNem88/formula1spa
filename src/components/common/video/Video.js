import React from 'react'
import ReactPlayer from 'react-player/youtube'
import styles from './video.module.css'
function Video({url}) {
   
    return (
        <div className={styles['player-wrapper']}>
            <ReactPlayer className={styles.player} width={'100%'} height={'100%'} url={url} light={true} controls={true} />
        </div>
    )
}

export default Video