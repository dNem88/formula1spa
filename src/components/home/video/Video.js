import React from 'react'
import ReactPlayer from 'react-player/youtube'
import styles from './video.module.css'
function Video(props) {
    let youtubeUrl = 'https://www.youtube.com/watch?v=aPytuKo87W8';
    return (
        <div className={styles['player-wrapper']}>
            <ReactPlayer className={styles.player} width={'100%'} height={'100%'} url={youtubeUrl} controls={true} muted={true}/>
        </div>
    )
}

export default Video