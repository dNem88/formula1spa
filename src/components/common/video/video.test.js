import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import Video
from './Video'

const mockProps = {
    url: 'test-url'
}

describe('test Video', () => {
    it('Snapshot test Video component', () => {
        let component = renderer.create(
            <MemoryRouter>
                <Video url={mockProps.url}/>
            </MemoryRouter>
        ).toJSON();
        expect(component).toMatchSnapshot()
        expect(component.type).toBe('div')
        expect(component.props.className)
.toBe('player-wrapper')   
        expect(component.children.length).toBe(1)
        expect(component.children[0].type).toBe('div')
        expect(component.children[0].props.style.width).toBe('100%')
        expect(component.children[0].props.style.height).toBe('100%')
        expect(component.children[0].props.className).toBe('player')
        
 })
})