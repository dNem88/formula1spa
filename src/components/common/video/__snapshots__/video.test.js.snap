// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Video Snapshot test Video component 1`] = `
<div
  className="player-wrapper"
>
  <div
    className="player"
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</div>
`;
